lib/spack/spack/test/cmd/config.py

```diff
@@ -399,7 +399,8 @@ def test_remove_list(mutable_empty_config):
 
 def test_config_add_to_env(mutable_empty_config, mutable_mock_env_path):
     env = ev.create('test')
-    with env:
+    env.write()  # We can only preserve comments on a written environment
```

Comment:
There are comments in the string we check (spack.environment.default_manifest has comments in it).

Reply:
OK: I think either

* the docstring should mention we are checking for preservation of comments
* we should just check the value of `config:dirty` after separately reading (vs. doing a string match)

I think the latter, particularly since this would depend on whether the default manifest YAML has comments, which is not clear from this test (and hopefully we have a test specifically about comments).